hw7/functional.py

```diff
@@ -0,0 +1,58 @@
+def sequential_map(*args):
+    container = list(args[-1])
+    functions = args[0:(len(args)-1)]
+    for function in functions:
+        container = list(map(function, container))
+    return container
+
+def consensus_filter(*args):
+    container = list(args[-1])
+    functions = args[0:(len(args)-1)]
+    for function in functions:
+        container = list(filter(function, container))
+    return container
+
+def conditional_reduce(fun1, fun2, container):
+    # Создадим функцию reduce вместо импорта из functools    
+    def reduce(function, iterable, initializer=None):
+        it = iter(iterable)
```

Comment:
Нейминг переменных, не понятно, что за `it`. Как вариант можно было хотя бы
```suggestion
        iterable = iter(iterable)
```
![image](https://user-images.githubusercontent.com/48160953/207434148-1dd60f48-e586-4922-85f2-8d04408d9032.png)